great_expectations/expectations/expectation.py

```diff
@@ -1736,6 +1736,12 @@ def _validate_metric_value_between(
                 except TypeError:
                     pass
 
+        elif isinstance(metric_value, datetime.datetime):
```

Comment:
@jcampbell This is correctly done.  Thanks for putting it in (and makes sense).  Since I will be doing work on the same method, I would like to cherry pick it into my pull request (and will let you know).  Thank you.